### PR TITLE
Issue #57 Adding the first meaningful Freshsales sample app

### PR DESCRIPTION
**Pull request by moniksha-murthy**

### CODE REVIEW

File: Freshsales/simple_request_app/app/app.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+function getLeadDetails(leadId) {
+    var headers = {"Authorization": "Token token=<%= (iparam.api_key) %>"};
+    var options = { headers: headers };
+    var url = "<%= (iparam.account) %>/api/leads/" + leadId;
+
+    client.request.get(url, options).then (function(data) {
+        displayLeadDetails(data.response);
+        showNotification('success', 'Lead info retrieved successfully');
+    }).catch(function(e) {
+        console.error('Exception - ', e);
```
Comment (ravirajthedeveloper) [inline Freshsales/simple_request_app/app/app.js]: Could you pls add a more descriptive error message to understand where the error comes from?

File: Freshsales/simple_request_app/app/app.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    var cellOne = newRow.insertCell(0);
+    cellOne.classList.add('bold');
+    var cellOneText = document.createTextNode(key);
+    cellOne.appendChild(cellOneText);
+
+    var cellTwo = newRow.insertCell(1);
+    var cellTwoText = document.createTextNode(value);
+    cellTwo.appendChild(cellTwoText);
+}
```
Comment (ravirajthedeveloper) [inline Freshsales/simple_request_app/app/app.js]: Add an empty line at the EOF at this file and the other files that are missing it.

File: Freshsales/simple_request_app/config/iparams.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+  "account": {
+    "display_name": "Freshsales acount url",
+    "description": "Please enter your Freshsales account url in the format https://domain.freshsales.io",
+    "type": "text",
+    "required": true
+  },
+  "api_key": {
```
Comment (ravirajthedeveloper) [inline Freshsales/simple_request_app/config/iparams.json]: Kindly mark this API key as secure.

Comment (moniksha-murthy) [reply Freshsales/simple_request_app/config/iparams.json]: done

File: Freshsales/simple_request_app/app/app.js (hunk 1 above)
Comment (moniksha-murthy) [reply Freshsales/simple_request_app/app/app.js]: done

File: Freshsales/simple_request_app/app/app.js (hunk 2 above)
Comment (moniksha-murthy) [reply Freshsales/simple_request_app/app/app.js]: done

File: Freshsales/simple_request_app/app/app.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+function getLeadDetails(leadId) {
+    var headers = {"Authorization": "Token token=<%= (iparam.api_key) %>"};
+    var options = { headers: headers };
+    var url = "<%= (iparam.account) %>/api/leads/" + leadId;
+
+    client.request.get(url, options).then (function(data) {
+        displayLeadDetails(data.response);
+        showNotification('success', 'Lead info retrieved successfully');
+    }).catch(function(e) {
+        console.error('Error occurred while retrieving lead details: ', e);
```
Comment (ravirajthedeveloper) [inline Freshsales/simple_request_app/app/app.js]: Could you pls add a showNotify to notify the user about the failure and why the don't show anything in the frontend with some action items like "try sometime later"?
Without this, the app would look like struck with no notification to the user in the UI.

Comment (moniksha-murthy) [reply Freshsales/simple_request_app/app/app.js]: I initially added it, but looks like Freshsales doesn't support interface api - error notification. https://developers.freshsales.io/docs/interface/#shownotification

Comment (moniksha-murthy) [reply Freshsales/simple_request_app/app/app.js]: So removed it after testing

Comment (ravirajthedeveloper) [reply Freshsales/simple_request_app/app/app.js]: Then it's fine. We will revisit it when the support for error type is added.

**Pull request merged**